app/services/conversation-channel.js

```diff
@@ -1,39 +1,29 @@
-import PhoenixSocket from 'phoenix/services/phoenix-socket';
 import { get } from '@ember/object';
-import { inject as service } from '@ember/service';
-import ENV from 'code-corps-ember/config/environment';
+import Service, { inject as service } from '@ember/service';
 
-export default PhoenixSocket.extend({
-  session: service(),
+export default Service.extend({
+  socket: service(),
   store: service(),
 
-  connect() {
-    if (!get(this, 'socket')) { // Only create one socket at a time
-      let token = get(this, 'session.session.authenticated.token');
-      this._super(ENV.SOCKETS_BASE_URL, {
-        params: { token }
-      });
-    }
-  },
-
   async join(conversation) {
-    await this.connect();
+    let socket = get(this, 'socket');
+
+    await socket.connect();
```

Comment:
Any failure state we need to be aware of?

Reply:
There are some defined in Phoenix's JS but none I think we _need_ to handle here right now.